### PR TITLE
Added SLURM functionality to CI

### PR DESCRIPTION
**Pull request by xchoo**
- Modified secret, .nengobones.yml file to use GCI network
- Added conda installs for pandoc, ffmpeg
- Updated remote-script to use NengoBones remote-script
  template
TEMPORARY CHANGES (until changes are merged into NengoBones)
- Add base-script, docs, and slurm-script to support SLURM
  jobs
- Updated remote-script after_scripts check for all docs
  scripts

**How has this been tested?**
Tested with CI. Tested locally.

**How long should this take to review?**
- Quick (less than 40 lines changed or changes are straightforward)

**Where should a reviewer start?**
Start with .nengobones.yml file

**Types of changes:**
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

**Checklist:**
- [x] I have read the **CONTRIBUTING.rst** document.
- [N/A] I have updated the documentation accordingly.
- [N/A] I have included a changelog entry.
- [x] I have tested this with all supported devices.
- [N/A] I have added tests to cover my changes.
- [x] I have run the test suite locally and all tests passed.

**Still to do:**

### CODE REVIEW

File: .templates/base_script.sh.template
Comment (bmorcos) [inline .templates/base_script.sh.template:23]: Looks like this part is the only change from the current base script, just for my own interest why was this required?

Comment (bmorcos) [inline .templates/base_script.sh.template:23]: Looks like this part is the only change from the current base script, just for my own interest why was this required?

File: .templates/docs.sh.template
Comment (bmorcos) [inline .templates/docs.sh.template:1]: I think it makes sense to put the `fpga_config` stuff in the `docs` script instead of the `remote-script` since it's related to this content and not the remote wrapper. In that case, once `nengo-bones` is updated we can just extend the `docs` script with the `fpga_config` part and we can use the default `nengo-bones` version of `remote-script` which should be nice and clean.

**Review by bmorcos — APPROVED** [review]
Are we expecting the `nengo-bones` PR to take a while or can we just wait for that to update this repo with SLURM?

Just had the one question and then the suggestion to move the `fpga_config` stuff to `docs`, but I'll leave that up to you.

File: .templates/docs.sh.template
Comment (xchoo) [reply .templates/docs.sh.template:1]: Oh... this `docs` template is here only because it uses the modified `base_script` template. When the slurm changes are merged into NengoBones, this file is not needed, we can use the default `docs` template.

File: .templates/base_script.sh.template
Comment (xchoo) [reply .templates/base_script.sh.template:23]: > Are we expecting the nengo-bones PR to take a while or can we just wait for that to update this repo with SLURM?

There is no timeframe on that.. and seeing as everyone is busy, it might take a while.

> Looks like this part is the only change from the current base script, just for my own interest why was this required?

The `eval` is needed to properly process the aliases. In the `slurm-script`, it creates aliases for any `wrapped_commands` 
(e.g., `alias sphinx-build="srun -pCI sphinx-build"`), and while the aliases work in the bash terminal, I found that I needed to do `eval <command>` to get it to be recognized within the `exe` function.

File: .templates/docs.sh.template
Comment (xchoo) [reply .templates/docs.sh.template:1]: I feel like we should leave the `docs` template alone (i.e., don't put any modifiers in it) so that people can still run it on their local machines with their own configs. Putting the `fpga_config` stuff in the `remote-script` allows us to do that.

Comment (bmorcos) [reply .templates/docs.sh.template:1]: Yea I understand that, hence "once nengo-bones is updated...". This is more a comment on the fact that I (subjectively) think it makes more sense to have the extra `fpga_config` content in the `docs` template instead of the `remote-script` template since it pertains to building the docs, not running remotely. Either way once updates are merged into `nengo-bones` we will need one of either `docs` or `remote-script` to be custom and I think it makes more sense for that to be `docs`.

File: .templates/base_script.sh.template
Comment (bmorcos) [reply .templates/base_script.sh.template:23]: You should be able to just `extend` the base script instead of copying the whole thing actually, then just override the header block. to keep things cleaner (even though it's temporary).

File: .templates/docs.sh.template
Comment (bmorcos) [reply .templates/docs.sh.template:1]: O sorry your second comment hadn't loaded when I typed this... That makes sense to me for leaving it in `remote-script`.

Comment (xchoo) [reply .templates/docs.sh.template:1]: I have a different opinion. With the `fpga_config` stuff in `remote-script` (because it really is only for the remote CI setup, you can build the docs by running the CI scripts directly (and you don't have to remember the exact `sphinx-build` command to do so). You can even call `.ci/docs.sh install` to install all of the necessary packages (except for `pandoc`) you need to do the docs build. If we put the `fpga_config` changes in the `docs.sh` template, it breaks this functionality since it's likely that the `fpga_config` in use would be the root `nengo-fpga` directory, and not in the home directory.

**Pull request merged**
